src/pluginManager.ts

```diff
@@ -144,18 +145,18 @@ export class PluginManager {
         log.info(`Loaded plugin: ${identifier}@${plugin.version}`);
       }
 
-      this.initializePlugin(plugin, identifier);
+      await this.initializePlugin(plugin, identifier);
 
       log.info("---");
-    });
+    }
 
     this.currentInitializingPlugin = undefined;
   }
 
-  public initializePlugin(plugin: Plugin, identifier: string): void {
+  public async initializePlugin(plugin: Plugin, identifier: string): Promise<void> {
     try {
       this.currentInitializingPlugin = plugin;
-      plugin.initialize(this.api); // call the plugin's initializer and pass it our API instance
+      await plugin.initialize(this.api); // call the plugin's initializer and pass it our API instance
```

Comment:
The await here has currently no effect(?), what was the intend here?

You said a homebridge plugin could now also export a async function.
Though looking into the `Plugin.initialize(api: API): void` (which is currently not a async function), the line `this.pluginInitializer(api);` isn't waited for(?), so seemingly async functions exported by the plugin aren't waited for, or am I missing something?

Reply:
Good catch. I missed that function. 
Rather than awaiting, I think it's better if we do `return ...` in `https://github.com/homebridge/homebridge/blob/master/src/plugin.ts#L203` and thus the new signature should be `Plugin.initialize(api: API): Promise<unknown>`.

Reply:
Updated. PTAL.